uniffi_bindgen/src/library_mode.rs

```diff
@@ -136,12 +137,13 @@ pub fn calc_cdylib_name(library_path: &Utf8Path) -> Option<&str> {
     None
 }
 
-fn find_sources<Config: BindingsConfig>(
+fn find_sources<T: BindingGenerator>(
+    generator: &T,
     cargo_metadata: &cargo_metadata::Metadata,
     library_path: &Utf8Path,
     cdylib_name: Option<&str>,
     config_file_override: Option<&Utf8Path>,
```

Comment:
Can you please elaborate here? I've some vague ideas for going further in a later patch and I'd like to understand your thoughts. Here in particular I don't think we have the path to the TOML yet - we use the cargo metadata to find potentially one TOML file per crate.

What I think I'd like to do is work out how to separate the `Config` from the `Generator` - this would mean everything could pass around a `dyn BindingGenerator` and a `dyn BindingConfig` which would simplify a number of things. However, the only way I can see to make that work is to lean in to `std::any::Any` and have the binding downcast from the `dyn BindingConfig` back into its concrete `Config`. Any thoughts on that?